native/spark-expr/src/math_funcs/abs.rs

```diff
@@ -0,0 +1,806 @@
+                                Err(_) => Err(arithmetic_overflow_error("Decimal128").into()),
+                            }
+                        }
+                        _ => Err(DataFusionError::Internal("Invalid data type".to_string())),
+                    }
+                }
+            }
+            DataType::Decimal256(precision, scale) => {
+                if !fail_on_error {
+                    let result =
+                        legacy_compute_op!(array, wrapping_abs, Decimal256Array, Decimal256Array)?;
+                    let result = result.with_data_type(DataType::Decimal256(*precision, *scale));
+                    Ok(ColumnarValue::Array(Arc::new(result)))
+                } else {
+                    // Need to pass precision and scale from input, so not using ansi_compute_op
+                    let input = array.as_any().downcast_ref::<Decimal256Array>();
+                    match input {
+                        Some(i) => {
+                            match arrow::compute::kernels::arity::try_unary(i, |x| {
+                                if x == i256::MIN {
+                                    Err(ArrowError::ArithmeticOverflow("Decimal256".to_string()))
+                                } else {
+                                    Ok(x.wrapping_abs()) // i256 doesn't define abs() method
+                                }
+                            }) {
+                                Ok(res) => Ok(ColumnarValue::Array(Arc::<
+                                    PrimitiveArray<Decimal256Type>,
+                                >::new(
+                                    res.with_data_type(DataType::Decimal256(*precision, *scale)),
+                                ))),
+                                Err(_) => Err(arithmetic_overflow_error("Decimal256").into()),
+                            }
+                        }
+                        _ => Err(DataFusionError::Internal("Invalid data type".to_string())),
+                    }
+                }
+            }
+            dt => exec_err!("Not supported datatype for ABS: {dt}"),
+        },
+        ColumnarValue::Scalar(sv) => match sv {
+            ScalarValue::Null
+            | ScalarValue::UInt8(_)
+            | ScalarValue::UInt16(_)
+            | ScalarValue::UInt32(_)
+            | ScalarValue::UInt64(_) => Ok(args[0].clone()),
+            ScalarValue::Int8(a) => a
+                .map(|v| match v.checked_abs() {
+                    Some(abs_val) => Ok(ColumnarValue::Scalar(ScalarValue::Int8(Some(abs_val)))),
+                    None => {
+                        if !fail_on_error {
+                            // return the original value
+                            Ok(ColumnarValue::Scalar(ScalarValue::Int8(Some(v))))
+                        } else {
+                            Err(arithmetic_overflow_error("Int8").into())
+                        }
+                    }
+                })
+                .unwrap(),
```

Comment:
What should be the behavior here if the argument is `ColumnarValue::Scalar(ScalarValue::Int8(None))` ?
Because currently the `.unwrap()` would panic.
IMO it should just return `ColumnarValue::Scalar(ScalarValue::Int8(None))`

Reply:
Thanks @martin-g. I have addressed this in the replacement PR https://github.com/apache/datafusion-comet/pull/2689